flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -804,6 +809,160 @@ protected static String getRefreshStatement(
         return insertStatement.toString();
     }
 
+    private ResultFetcher callAlterMaterializedTableAsQueryOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            AlterMaterializedTableAsQueryOperation op) {
+        ObjectIdentifier tableIdentifier = op.getTableIdentifier();
+        CatalogMaterializedTable oldMaterializedTable =
+                getCatalogMaterializedTable(operationExecutor, tableIdentifier);
+
+        if (CatalogMaterializedTable.RefreshMode.FULL == oldMaterializedTable.getRefreshMode()) {
```

Comment:
I think we forgot one important logic: explain the select query and ensure it can compile successfully in the corresponding refresh mode when creating and modifying a query. We should do more pre-check actions before creating or altering a query. Can you help create one Jira issue to track it?

Reply:
I created https://issues.apache.org/jira/browse/FLINK-37066